testing/node_fetcher.ts

```diff
@@ -178,12 +159,6 @@ export async function nodeFetcher(options: BaseFetcherOptions): Promise<FetcherF
     body: resultBody,
   };
 
-  if (simple) {
-    if (!response.ok) {
-      throw new StatusCodeError(response.status, resultBody, options, fullResponse);
-    }
-  }
```

Comment:
node fetcher doesn't throw with status code by itself. this is done by `handleFetcherStatusError`. 

while it's not a big deal, this particular implementation uses a separate instance of `StatusCodeError` that confuses error handling. I figured it's easier to just remove the whole thing.

Reply:
This code is similar to what's in production, so if a change is needed here then a corresponding change may be needed in `coda`

Reply:
The node-fetch in production did throw but the fetcher service doesn't. it will catch the status code error and return a regular fetcher response. https://github.com/kr-project/experimental/blob/8a9d698d9af8e89822b7a8b51981bf3ad146be60/modules/server/fetcher-server/private/fetcher_service.ts#L352